<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Sign Language Translator</string>
  </property>
  <property name="styleSheet">
   <string notr="true">QMainWindow {
    background-color: #f5f5f5;
}
QGroupBox {
    border: 1px solid #ddd;
    border-radius: 5px;
    margin-top: 10px;
    padding-top: 15px;
    font-size: 16px;
    font-weight: bold;
    color: #333;
}
#cameraGroup::title, #transcriptionGroup::title {
    color: white;
    background-color: #4CAF50;
    padding: 4px 10px;
    border-radius: 3px;
    margin-left: 10px;
    subcontrol-origin: margin;
}
QPushButton {
    background-color: #4CAF50;
    border: none;
    color: white;
    padding: 10px 20px;
    text-align: center;
    text-decoration: none;
    font-size: 14px;
    margin: 4px 2px;
    border-radius: 4px;
    min-width: 120px;
}
QPushButton:hover {
    background-color: #45a049;
}
QTextEdit {
    background-color: black;
    border: 1px solid #ddd;
    border-radius: 4px;
    padding: 8px;
    font-size: 25px;
    color: white;
}
QLabel {
    font-weight: bold;
    font-size: 25px;
    color: #333;
}
QGraphicsView {
    background-color: #333;
}

/* Settings Group Box */
#settingsGroup {
    background-color: #4CAF50;
    border: 2px solid #45a049;
}

#settingsGroup::title {
    color:  #f3f9f4 ;
    background-color: #4CAF50;
    padding: 4px 10px;
    border-radius: 3px;
    margin-left: 10px;
    subcontrol-origin: margin;
}

/* Confidence Threshold Section */
#confidenceLabel {
    font-size: 16px;
    color: #333;
    font-weight: bold;
    padding: 5px;
    background-color: #e8f5e9;
    border-radius: 4px;
    border: 1px solid #c8e6c9;
}

#confidenceValueLabel {
    font-weight: bold;
    font-size: 16px;
    color: #000000;
    background-color: #ffffff;
    border-radius: 4px;
    padding: 4px 8px;
    border: 1px solid #bdc3c7;
    min-width: 50px;
    text-align: center;
}

/* Timer Labels */
#wordTimerLabel, #letterTimerLabel {
    font-size: 14px;
    color: #333;
    font-weight: bold;
    padding: 4px 8px;
    background-color: #e8f5e9;
    border-radius: 4px;
    border: 1px solid #c8e6c9;
    margin-right: 5px;
}

/* Slider Styling */
QSlider {
    min-height: 20px;
}
QSlider::groove:horizontal {
    border: 1px solid #bbb;
    background: #f0f0f0;
    height: 8px;
    border-radius: 4px;
}
QSlider::sub-page:horizontal {
    background:    #0cde5c;
    border: 1px solid  #45a049;
    height: 8px;
    border-radius: 4px;
}
QSlider::handle:horizontal {
    background: #fff;
    border: 1px solid #ccc;
    width: 16px;
    margin: -4px 0;
    border-radius: 8px;
}
QSlider::handle:horizontal:hover {
    background: #f5f5f5;
    border: 1px solid #bbb;
}

/* Line Edit Styling */
QLineEdit {
    border: 1px solid #95a5a6;
    border-radius: 4px;
    padding: 4px;
    font-size: 14px;
    font-weight: bold;
    color: #000000;
    background: white;
    selection-background-color: #4CAF50;
    min-width: 50px;
}
QLineEdit:focus {
    border: 2px solid #4CAF50;
}

/* Vertical Line Separator */
QFrame[frameShape="4"] {  /* VLine */
    color: #c8e6c9;
    background-color: #c8e6c9;
}
</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="mainVerticalLayout">
    <property name="spacing">
     <number>20</number>
    </property>
    <!-- Main Content Area -->
    <item>
     <layout class="QHBoxLayout" name="contentLayout">
      <property name="spacing">
       <number>20</number>
      </property>
      <!-- Camera Feed -->
      <item>
       <widget class="QGroupBox" name="cameraGroup">
        <property name="title">
         <string>Camera Feed</string>
        </property>
        <layout class="QVBoxLayout" name="cameraLayout">
         <item>
          <widget class="QGraphicsView" name="videoView">
           <!-- Remove fixed minimum/maximum sizes -->
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>1</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      
      <!-- Transcription Box -->
      <item>
       <widget class="QGroupBox" name="transcriptionGroup">
        <property name="title">
         <string>Transcription</string>
        </property>
        <layout class="QVBoxLayout" name="transcriptionLayout">
         <item>
          <widget class="QTextEdit" name="transcriptionTextBox">
           <!-- Remove fixed minimum/maximum sizes -->
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>1</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="verticalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="horizontalScrollBarPolicy">
            <enum>Qt::ScrollBarAlwaysOff</enum>
           </property>
           <property name="font">
            <font>
             <family>Arial</family>
             <pointsize>12</pointsize>
            </font>
           </property>
           <property name="readOnly">
            <bool>false</bool>
           </property>
           <property name="lineWrapMode">
            <enum>QTextEdit::NoWrap</enum>
           </property>
           <property name="acceptRichText">
            <bool>false</bool>
           </property>
           <property name="overwriteMode">
            <bool>false</bool>
           </property>
           <property name="tabChangesFocus">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>

    <!-- Recognition Controls Box -->
    <item>
     <widget class="QGroupBox" name="settingsGroup">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Sign Recognition Controls</string>
      </property>
      <layout class="QHBoxLayout" name="settingsLayout">
       <property name="spacing">
        <number>15</number>
       </property>
       <item>
        <layout class="QVBoxLayout" name="confidenceLayout">
         <property name="spacing">
          <number>5</number>
         </property>
         <item>
          <widget class="QLabel" name="confidenceLabel">
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
           <property name="text">
            <string>Confidence Threshold</string>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="confidenceControlLayout">
           <property name="spacing">
            <number>5</number>
           </property>
           <item>
            <widget class="QSlider" name="confidenceSlider">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>99</number>
             </property>
             <property name="value">
              <number>99</number>
             </property>
             <property name="singleStep">
              <number>1</number>
             </property>
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="minimumSize">
              <size>
               <width>150</width>
               <height>20</height>
              </size>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="confidenceValueLabel">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="text">
              <string>0.99</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       
       <item>
        <widget class="QFrame" name="verticalLine">
         <property name="frameShape">
          <enum>QFrame::VLine</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
         <property name="lineWidth">
          <number>1</number>
         </property>
        </widget>
       </item>
       
       <item>
        <layout class="QVBoxLayout" name="timersLayout">
         <property name="spacing">
          <number>5</number>
         </property>
         <item>
          <layout class="QHBoxLayout" name="wordTimerLayout">
           <property name="spacing">
            <number>5</number>
           </property>
           <item>
            <widget class="QLabel" name="wordTimerLabel">
             <property name="text">
              <string>Word Confirmation Timer</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="wordTimerEdit">
             <property name="maximumSize">
              <size>
               <width>50</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="text">
              <string>2</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="letterTimerLayout">
           <property name="spacing">
            <number>5</number>
           </property>
           <item>
            <widget class="QLabel" name="letterTimerLabel">
             <property name="text">
              <string>Double Letter Timer</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="letterTimerEdit">
             <property name="maximumSize">
              <size>
               <width>50</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
             <property name="text">
              <string>3</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    
    <!-- Button Controls -->
    <item>
     <layout class="QHBoxLayout" name="buttonLayout">
      <property name="spacing">
       <number>20</number>
      </property>
      <item>
       <widget class="QPushButton" name="cameraOnButton">
        <property name="text">
         <string>Start Camera</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="cameraOffButton">
        <property name="text">
         <string>Stop Camera</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="toggleLandmarksButton">
        <property name="text">
         <string>Toggle Landmarks</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="clearButton">
        <property name="text">
         <string>Clear Text</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="exitButton">
        <property name="text">
         <string>Exit</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>